flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/PreviousAllocationSchedulingStrategy.java

```diff
@@ -48,35 +50,68 @@ private PreviousAllocationSchedulingStrategy() {}
 	@Override
 	public <IN, OUT> OUT findMatchWithLocality(
 			@Nonnull SlotProfile slotProfile,
-			@Nonnull Stream<IN> candidates,
-			@Nonnull Function<IN, SlotContext> contextExtractor,
+			@Nonnull Supplier<Stream<IN>> candidates,
```

Comment:
Not sure whether `Supplier<Stream<IN>>` is the best construct to be able to iterate over a collection of `IN` multiple times. I think either `Collection` or `Iterable` serve a better purpose here.

Reply:
I think both ways have their pros and cons, if we use a supplier for stream, we can benefit from the lazy evaluation of streaming. In one case for scheduling where we need to iterate the stream twice, havign a collection would be better. I feel ok with both variants if you think `Collection` is preferable.

Reply:
Let's keep it like it is.